components/sup/src/util/cli.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate tabwriter;
+
+use std::io::{self, Write, Stdout};
+use self::tabwriter::TabWriter;
+
+pub struct TableWriter {
```

Comment:
I'm not sure that a `TableWriter` struct necessarily buys us much here, and isn't quite general purpose (for example, nothing requires that you call `set_titles` first, and nothing prevents you from calling it after you've already added some content lines).  The `panic!` call is also problematic, as you pointed out.

It may just be easier to remove this struct and just assemble the table directly in `print_statuses`.

Reply:
The idea here was to finish the job later actually ;) needed a first look on the overall changes. I effectively want to forbid titles after lines. And to create he basis of a utility struct that could be used in other places of the app.

Reply:
I think the simpler way is good for now. We can employ the "Rule of 3" for further abstraction: when we find ourselves implementing the same pattern the third time, we should have a better idea what the real abstraction should be.

Reply:
Good for me! I'll change this then ;)